worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java

```diff
@@ -424,29 +414,33 @@ public Void call() throws Exception {
         }
     }
 
-    private static class SchematicShareTask extends SchematicOutputTask<URL> {
+    private static class SchematicShareTask implements Callable<URL> {
+        private final Actor actor;
+        private final ClipboardHolder holder;
         private final String name;
+        private final ClipboardShareDestination destination;
+        private final ClipboardFormat format;
 
-        SchematicShareTask(Actor actor, ClipboardFormat format, ClipboardHolder holder, String name) {
-            super(actor, format, holder);
+        SchematicShareTask(Actor actor,
+                           ClipboardHolder holder,
+                           ClipboardShareDestination destination,
+                           ClipboardFormat format,
+                           String name) {
+            this.actor = actor;
+            this.holder = holder;
             this.name = name;
+            this.destination = destination;
+            this.format = format;
         }
 
         @Override
         public URL call() throws Exception {
-            ByteArrayOutputStream baos = new ByteArrayOutputStream();
-            try {
-                writeToOutputStream(baos);
-            } catch (Exception e) {
-                throw new CommandException(TextComponent.of(e.getMessage()), e, ImmutableList.of());
-            }
-
-            EngineHubPaste pasteService = new EngineHubPaste();
             PasteMetadata metadata = new PasteMetadata();
             metadata.author = this.actor.getName();
             metadata.extension = "schem";
```

Comment:
should this be based on the format field instead? is our PasteMetadata really something that should be generically exposed to the ClipboardShareDestination class? seems more specific to EH paste.

Reply:
Hmm, fair - I feel most of this metadata makes sense from a general perspective, but extension maybe less so.